### PR TITLE
Has Archive Functionnality

### PR DESCRIPTION
**Pull request by D-Raoul**

### CODE REVIEW

File: app/Espo/Core/SelectManagers/Base.php [hunk 1]
```diff
@@ -858,6 +867,17 @@ public function getSelectParams(array $params, $withAcl = false, $checkWherePerm
 
         $this->q($params, $result);
 
+        // check if entity has hasArchive activated
+        if ($this->metadata->get("scopes")[$this->entityType]['hasArchive']) {
```
Comment (rratsun) [inline app/Espo/Core/SelectManagers/Base.php:871]: $this->metadata->get("scopes")[$this->entityType]['hasArchive'] should be changed to

$this->metadata->get(['scopes', $this->entityType, 'hasArchive'])

File: app/Espo/Listeners/Metadata.php [hunk 2]
```diff
@@ -445,6 +447,29 @@ protected function setAssetThumbSize(array $data): array
...
+
+                if (!isset($data['entityDefs'][$scope]['fields']['isArchived']['notNull'])) {
+                    $data['entityDefs'][$scope]['fields']['isArchived']['notNull'] = true;
+                }
+            }
+        }
+
+        foreach ($data['entityDefs'] as $entity => $row) {
+            if (isset($row['fields']['isArchived']['type']) && $row['fields']['isArchived']['type'] == 'bool') {
+                $data['clientDefs'][$entity]['filterList'][] = 'withArchived';
```
Comment (rratsun) [inline app/Espo/Listeners/Metadata.php:466]: Why 'filterList'? It should be boolFilterList

File: client/modules/treo-core/src/views/site/header.js [hunk 3]
```diff
@@ -48,15 +48,15 @@ Espo.define('treo-core:views/site/header', 'class-replace!treo-core:views/site/h
...
-                $.ajax('data/publicData.json?silent=true&time=' + $.now(), {local: true}).done(response => {
-                    Backbone.Events.trigger('publicData', response);
-                    if (response.dataTimestamp) {
-                        $.each(response, (k, v) => {
-                            localStorage.setItem('pd_' + k, v);
-                        });
-                        this.isNeedToReloadPage();
-                    }
-                });
+                // $.ajax('data/publicData.json?silent=true&time=' + $.now(), {local: true}).done(response => {
```
Comment (rratsun) [inline client/modules/treo-core/src/views/site/header.js:51]: What is it?

File: app/Espo/Listeners/Metadata.php (hunk 2 above)
Comment (D-Raoul) [reply app/Espo/Listeners/Metadata.php:466]: Alex changed the requirements, now the filter is shown in the primary filter list

File: client/modules/treo-core/src/views/site/header.js (hunk 3 above)
Comment (D-Raoul) [reply client/modules/treo-core/src/views/site/header.js:51]: sorry, i commented this to not have all those request in the debugger, i revert that immediatly

File: app/Espo/Core/SelectManagers/Base.php (hunk 1 above)
Comment (D-Raoul) [reply app/Espo/Core/SelectManagers/Base.php:871]: OK

**Pull request merged**